Add xunit perf test project.

codito opened the pull request:
Extract common dependencies to a props.

Commit:
Add xunit perf test project.

Extract common dependencies to a props.

msftclas commented:
Hi __@codito__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <p>
        It looks like you're a Microsoft contributor (Arun Mahapatra). If you're full-time, we DON'T require a Contribution License Agreement. If you are a vendor, please DO sign the electronic Contribution License Agreement. It will take 2 minutes and there's no faxing! https://cla.microsoft.com.
    </p>

TTYL, MSBOT;

AbhitejJohn commented:
@codito : Should we start moving some of these assets out of TestPlatform.sln to reduce build time?

Faizan2304 commented on scripts/build/TestPlatform.Settings.targets:
```diff
@@ -31,6 +31,9 @@
   </PropertyGroup> 
 
+  <!-- Package dependency versions -->
+  <Import Project="$(MSBuildThisFileDirectory)TestPlatform.Dependencies.props" />
```
I am not able to find TestPlatform.Dependencies.props. From where we are getting this.

codito replied:
Added now. It was ignore by gitignore earlier, fixed that as well.

codito commented:
@AbhitejJohn we don't want to differentiate VS build vs command line. TestPlatform.sln is built everywhere. These assets are required for running Performance Tests, it is an overhead to require running build.cmd if someone wants to run perf tests.

Commit:
Remove props from gitignore. Add Dependencies file.

smadala commented on test/TestAssets/SimpleTestProject/SimpleTestProject.csproj:
```diff
@@ -1,19 +1,22 @@
     </PackageReference>
     <PackageReference Include="MSTest.TestAdapter">
-      <Version>1.1.6-preview</Version>
+      <Version>$(MSTestAdapterVersion)</Version>
```
This should be sync with  [acceptance test reference](https://github.com/Microsoft/vstest/blob/master/test/Microsoft.TestPlatform.TestUtilities/IntegrationTestBase.cs#L31).
Is there way we can read reference from `prop` file directly in acceptance or we have to sync both manually always?

codito replied:
Good catch. We should pull that as a $env variable? Since integration are a child process of test.cmd?

Faizan2304 replied:
+1

smadala reviewed (approved):
Nice cleanup 👍.

AbhitejJohn commented:
@codito: Was referring to pulling it into another solution and build.cmd would build these two. But yeah, if it isn't adding much time no point. Just wanted to see if we can reduce build time for normal development.

Commit:
Merge remote-tracking branch 'upstream/master' into xunit-perf-asset

codito committed:
Merge branch 'master' into xunit-perf-asset

Commit:
Add package id for OM to prefer project reference.

Commit:
Add two flags LocalizeResources and LocalizedBuild.

Don't include all resx files if LocalizedBuild is false to speed up
build.

Commit:
Pick up MSTestAdapterPath from dependency props file in test.ps1.

Commit:
Add chutzpah as a dependency.
Read dependencies from props file in integration tests.
Publish testhost to package directory for net46 since it is included in the runner package.
